google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -41,6 +58,11 @@ def set_experiment(self, experiment: str):
             schema_version=constants.SCHEMA_VERSIONS[constants.SYSTEM_EXPERIMENT],
             metadata=constants.EXPERIMENT_METADATA,
         )
+        if context.schema_title != constants.SYSTEM_EXPERIMENT:
+            raise ValueError(
+                f"Experiment name {experiment} has been used to create other type of resources in this MetadataStore, "
```

Comment:
For these error messages, I would consider also providing the type of the retrieved context.

Reply:
Added the schema_title into the error string.